### DISCUSSION TITLE
[Bug]: API became slow

### DISCUSSION THREAD

**Discussion by ArefJafari** [acheong08/ChatGPT#1063] [General]
### Is there an existing issue for this?

- [X] I have searched the existing issues and checked the recent builds/commits

### What happened?

Since this morning, the speed of retrieving responses reduced. My experience is 10x slower than last Friday. I also upgrade the API but it didn't solve the problem.

### Steps to reproduce the problem

1. Go to ....
2. Press ....
3. ...

### What should have happened?

It was working much faster before.

### Version where the problem happens

3.0.2

### What is your operating system ?

Linux

### Command Line Arguments

```Shell
No
```

### Console logs

```Shell
None
```

**Comment by ghaddarAbs** [comment]
same issue here. I think it's related to this problem #1047 , @acheong08 after you removed the proxy the gateway error disappeared but now it has become 5 times slower than before.

**Comment by noseon** [comment]
he could release a proxy to run on Vercel so whoever wants to put it on this site and use their own proxy

**Comment by 7trail** [comment]
I can attest to this as well. I'm unsure what the proxy issues have been exactly (although I believe it was a DDOS?), but I have noticed a significant speed decline since last Friday.

**Comment by acheong08** [comment]
I removed proxy and used the official endpoint directly. The fact that it's slow seems to imply that proxy makes it faster? I'll try adding the proxy back with more servers load balancing and see if there are any speed improvements

**Comment by acheong08** [comment]
I just swapped out the endpoint with proxy and it somehow became faster. Magic lol

**Comment by ghaddarAbs** [comment]
I can confirm from two minutes of running that the speed returned as before, thanks @acheong08

**Reply by acheong08** [reply]
It's weird. Using the bypass endpoint directly is slow than through a proxy that uses the same thing?

**Comment by acheong08** [comment]
https://github.com/acheong08/ChatGPT/releases/tag/3.1.5
